apps/health_campaign_field_worker_app/lib/pages/beneficiary/deliver_intervention.dart

```diff
@@ -886,24 +890,31 @@ class _DeliverInterventionPageState
               )),
         ],
       ),
-      _quantityDistributedKey: FormArray<int>([
+      _quantityDistributedKey: FormArray<double>([
         ..._controllers.map(
-          (e) => FormControl<int>(
+          (e) => FormControl<double>(
             value: _controllers.indexOf(e) < productVariants.length
                 ? productVariants
                     .elementAt(_controllers.indexOf(e))
                     .quantity!
-                    .toInt()
+                    .toDouble()
                 : 0,
             validators: [Validators.min(1)],
           ),
         ),
       ]),
-      _quantityWastedKey: FormArray<String>([
+      _quantityWastedKey: FormArray<double>([
+        ..._controllers.map(
+          (e) => FormControl<double>(value: 0),
+        ),
+      ]),
+      _quantityWastedRadioKey: FormArray<KeyValue>([
         ..._controllers.map(
-          (e) => FormControl<String>(),
+          (e) =>
+              FormControl<KeyValue>(value: KeyValue('CORE_COMMON_NO', false)),
```

Comment:
use from i18 constants

Reply:
here actually We need to send the code  because  in the radiobuttonlist it is used to  translate the label of each radio